crates/chain/src/tx_graph.rs

```diff
@@ -644,19 +644,25 @@ impl<A: Anchor> TxGraph<A> {
 
     /// Get the position of the transaction in `chain` with tip `chain_tip`.
     ///
+    /// This method is like [`try_get_chain_position`] except it restricts the
+    /// chain to a custom tip. The tip doesn't even need to be in the same chain as the current tip.
+    /// You can use this to find information about a point in the past or on a fork if your
+    /// chain oracle supports that.
+    ///
     /// If the given transaction of `txid` does not exist in the chain of `chain_tip`, `None` is
     /// returned.
     ///
     /// # Error
     ///
     /// An error will occur if the [`ChainOracle`] implementation (`chain`) fails. If the
-    /// [`ChainOracle`] is infallible, [`get_chain_position`] can be used instead.
+    /// [`ChainOracle`] is infallible, [`get_subchain_position`] can be used instead.
     ///
-    /// [`get_chain_position`]: Self::get_chain_position
-    pub fn try_get_chain_position<C: ChainOracle>(
+    /// [`try_get_chain_position`]: Self::try_get_chain_position
+    /// [`get_subchain_position`]: Self::get_subchain_position
+    pub fn try_get_subchain_position<C: ChainOracle>(
         &self,
         chain: &C,
-        chain_tip: BlockId,
+        chain_tip: Option<BlockId>,
```

Comment:
Same as my comment above, why having an Option here?

Reply:
After a deeper look, I think we should keep this `Option` for 2 reasons: for consistency between methods and because it is convenient to check if the `chain_tip` is `None` in just one place instead of having to check every time before calling the method (the `ChainOracle` give an `Option`). If you think that is still the case to just pass the `BlockId` let me know and I will do it.

Reply:
> for consistency between methods

My point was that most (all?) of these methods should have a `BlockId` instead of an `Option`, I just commented a couple of them for brevity. 


I see that it's convenient to have an Option here, but at the same time it makes the API a bit more difficult to understand. 

Eventually it all comes down to personal preference, if you want to keep an `Option` here it's fine, but I think the decision should at least be documented properly

Reply:
I think I agree with @danielabrozzoni here. It causes friction to call it with the output of `get_chain_tip` but that friction is born by us with this PR mostly so +1.

Reply:
I get it, I'll fix the problem this weekend.